hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,105 @@
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class HrTimesheetSheet(models.Model):
+    _inherit = "hr_timesheet.sheet"
+
+    def _compute_attendance_count(self):
+        """Compute total number of attendance records
+        linked to current timesheet"""
+
+        for attendances in self:
+            attendances.attendance_count = len(attendances.attendances_ids)
+
+    @api.depends(
+        "timesheet_ids",
+        "timesheet_ids.unit_amount",
+        "attendances_ids",
+        "attendances_ids.check_in",
+        "attendances_ids.check_out",
+        "attendances_ids.employee_id",
+    )
+    def _compute_attendance_time(self):
+        """Compute total attendance time and
+        difference in total attendance-time
+        and timesheet-entry"""
+
+        current_date = datetime.now()
+        for sheet in self:
+            atte_without_checkout = sheet.attendances_ids.filtered(
+                lambda attendance: not attendance.check_out
+            )
+            atte_with_checkout = sheet.attendances_ids - atte_without_checkout
+            total_time = sum(atte_with_checkout.mapped("worked_hours"))
+            for attendance in atte_without_checkout:
+                delta = current_date - attendance.check_in
+                total_time += delta.total_seconds() / 3600.0
+            sheet.total_attendance = total_time
+
+            # calculate total difference
+            total_working_time = sum(sheet.mapped("timesheet_ids.unit_amount"))
+            sheet.total_difference = total_time - total_working_time
+
+    total_attendance = fields.Float(
+        compute="_compute_attendance_time",
+    )
+    total_difference = fields.Float(
+        compute="_compute_attendance_time",
+        string="Difference",
+    )
+    attendances_ids = fields.One2many(
+        comodel_name="hr.attendance", inverse_name="sheet_id", string="Attendances"
+    )
+    attendance_state = fields.Selection(
+        related="employee_id.attendance_state", string="Current Status"
+    )
+    attendance_count = fields.Integer(compute="_compute_attendance_count")
+
+    def attendance_action_change(self):
+        """Call attendance_action_change to
+        perform Check In/Check Out action
+        Returns last attendance record"""
+
+        return self.employee_id._attendance_action_change()
```

Comment:
By having this button in the timesheet, there could perhaps be a case where the user does two clicks in a fast way, then what would happen is that the check-in and check-out times are the same. 

If this happens, the attendance gets blocked as there is an Odoo standard check which verifies that the attendance check-in time is strictly minor than the check-out time (https://github.com/odoo/odoo/blob/15.0/addons/hr_attendance/models/hr_attendance.py#L78).

I don't know how should we address this, or perhaps skip it as it is an unlikely case. Any thoughts on this?

Reply:
@JasminSForgeFlow  was this comment attended somehow? Perhaps we can just add a note in the readme as known issues. WDYT?

Reply:
@AaronHForgeFlow yes I think we can add it in readme not as known issue.
Let me add it.

Reply:
@GuillemCForgeFlow are you fine with this?

Reply:
@JasminSForgeFlow @AaronHForgeFlow Yes, I agree, we can leave it as a known issue.

Reply:
Thank you :) Good for approval then? :pray: